src/components/search/map/marker.tsx

```diff
@@ -56,8 +58,25 @@ export const Marker = ({
     marker.setFocusHandler((focused: boolean) => onFocus(focused, id));
     marker.setHoverHandler((hovered: boolean) => onHover(hovered, id));
 
+    // Add the pin to the pinStore if it is not a cluster marker
+    const isClusterMarker = id.includes("cluster-");
+    if (cluster && cluster["setPinStore"] && !isClusterMarker) {
```

Comment:
Not sure I follow completely, is there a case where `cluster` is not null but doesn't have a `setPinStore` property?